### PR TITLE
Auction approval request to C2

### PR DESCRIPTION
**Pull request by jessieay**
* Closes https://github.com/18F/micropurchase/issues/771
* Admin can create C2 request via button
* Request approval is checked once per hour
* Only after request is approved can auction be published
* When auction is accepted, C2 proposal is updated with winner info

screen shot 2016-07-05 at 4 02 53 pm

### CODE REVIEW

File: app/services/construct_cap_attributes.rb [hunk 1]
```diff
@@ -16,13 +16,11 @@ def perform
...
         product_name_and_description: product_name_and_description,
         justification: justification,
-        link_to_product: link_to_product,
         cost_per_unit: cost_per_unit,
         quantity: ConstructCapAttributes::QUANTITY,
         recurring: ConstructCapAttributes::RECURRING,
         date_requested: date_requested,
-        urgency: ConstructCapAttributes::URGENCY,
-        additional_info: additional_info
+        urgency: ConstructCapAttributes::URGENCY
```
Comment (harrisj) [inline app/services/construct_cap_attributes.rb]: Is there a specific reason those fields were removed here? Not complaining, just wondering.

File: app/services/construct_cap_attributes.rb [hunk 2]
```diff
@@ -41,33 +39,18 @@ def product_name_and_description
...
-    auction.delivery_url
-  end
-
   def justification
     "Tock line item: #{auction.billable_to}"
   end
 
   def cost_per_unit
-    winning_bid.amount
+    auction.start_price
```
Comment (harrisj) [inline app/services/construct_cap_attributes.rb]: So we bill the auction start price, regardless of how much it actually sold for?

Comment (harrisj) [inline app/services/construct_cap_attributes.rb]: Oh I see. We need to create a proposal and that is the maximum price possible...

File: app/services/update_auction.rb [hunk 3]
```diff
@@ -56,27 +55,15 @@ def updating_ended_at?
   def perform_approved_auction_tasks
     if auction_accepted? && auction.accepted_at.nil?
       auction.accepted_at = Time.current
-    end
-
-    if should_create_cap_proposal?
-      CreateCapProposalJob.perform_later(auction.id)
+      UpdateCapProposalJob.perform_later(auction.id)
```
Comment (harrisj) [inline app/services/update_auction.rb]: Much nicer

File: app/services/update_cap_attributes.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      }
+    }
+  end
+
+  private
+
+  attr_reader :auction
+
+  def cost_per_unit
+    winning_bid.amount
```
Comment (harrisj) [inline app/services/update_cap_attributes.rb]: Nice, I see

File: app/services/construct_cap_attributes.rb (hunk 1 above)
Comment (harrisj) [inline app/services/construct_cap_attributes.rb]: Okay, I see. You add these fields when we update the CAP proposal

**Comment by harrisj** [comment]
This looks good, but it looks like it needs to be rebased. I also think we could get prettier buttons by moving them to a separate div and checking the font sizes (let's make them non-outlines too), but I don't know CSS

**Comment by adelevie** [comment]
Looks great to me once rebase is done 🌲

File: app/services/construct_cap_attributes.rb (hunk 2 above)
Comment (jessieay) [inline app/services/construct_cap_attributes.rb]: yeah, this is part of the functionality change we needed. The highest price needs to be included for approval before the auction runs, but the proposal is updated with real bid amount for purchase step on C2. C2 tracks all changes (a comment is added to the proposal after each update) so the changes will be recorded on their side.

**Comment by jessieay** [comment]
@harrisj rebased!

**Comment by jessieay** [comment]
@adelevie rebased!

**Comment by harrisj** [comment]
LGTM. Merging.

**Pull request merged**